src/calendar/grid/index.tsx

```diff
@@ -151,10 +168,9 @@ export default function Grid({
                 <td
                   key={`${weekIndex}:${dateIndex}`}
                   ref={tabIndex === 0 ? focusedDateRef : undefined}
-                  role="button"
                   tabIndex={tabIndex}
-                  aria-label={dayAnnouncement}
-                  aria-current={isSelected ? 'date' : undefined}
+                  aria-current={isDateOnSameDay ? 'date' : undefined}
+                  aria-selected={isSelected}
                   aria-disabled={!isEnabled}
                   onClick={onClick}
```

Comment:
In Safari+VO the "clickable" announcement is also done for disabled days when using table navigation.
Probably need to attach the onClick handler conditionally like `{ isEnabled ? () => onSelectDate(date) : undefined }` and add a test to ensure the handler is not added for disabled days.